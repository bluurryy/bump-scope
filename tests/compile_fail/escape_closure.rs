#![cfg_attr(feature = "nightly-allocator-api", feature(allocator_api))]
use bump_scope::Bump;

fn escape_closure(bump: &mut Bump) {
    let mut escapee = None;

    bump.scoped(|scope| {
        escapee = Some(scope.alloc("escape?"));
    });

    dbg!(escapee);
}

fn main() {}
