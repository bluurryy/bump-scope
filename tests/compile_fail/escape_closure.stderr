error[E0521]: borrowed data escapes outside of closure
 --> tests/compile_fail/escape_closure.rs:7:5
  |
4 |   let mut escapee = None;
  |       ----------- `escapee` declared here, outside of the closure body
5 |
6 |   bump.scoped(|scope| {
  |                ----- `scope` is a reference that is only valid in the closure body
7 |     escapee = Some(scope.alloc("escape?"));
  |     ^^^^^^^ `scope` escapes the closure body here
